File: Backend/dependency/inquiry/src/main/java/com/mhp/coding/challenges/dependency/inquiry/InquiryService.java
package com.mhp.coding.challenges.dependency.inquiry;

import com.mhp.coding.challenges.dependency.notifications.EmailHandler;
import com.mhp.coding.challenges.dependency.notifications.PushNotificationHandler;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class InquiryService {

    private static final Logger LOG = LoggerFactory.getLogger(InquiryService.class);

    private Inquiry inquiry;

    @Autowired
    private EmailHandler emailHandler;

    @Autowired
    private PushNotificationHandler pushNotificationHandler;

    public void create(final Inquiry inquiry) {
        this.inquiry = inquiry;
        LOG.info("User sent inquiry: {}", inquiry);
        emailHandler.sendEmail(inquiry);
        pushNotificationHandler.sendNotification(inquiry);
    }



}

File: Backend/dependency-dotnet/MHP.CodingChallenge.Backend.Dependency.Inquiry/InquiryService.cs
using System;
using MHP.CodingChallenge.Backend.Dependency.Inquiry.Interfaces;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Logging;


namespace MHP.CodingChallenge.Backend.Dependency.Inquiry
{
    public class InquiryService
    {
        private IEmailHandler EmailHandler;
        private IPushNotificationHandler NotificationHandler;
        public InquiryService(IEmailHandler emailHandler, IPushNotificationHandler pushNotificationHandler)
        {
            EmailHandler = emailHandler;
            NotificationHandler = pushNotificationHandler;
        }
        public void Create(Inquiry inquiry)
        {
            EmailHandler.SendEmail(inquiry);
            NotificationHandler.SendNotification(inquiry);
            Console.WriteLine(string.Format("User sent inquiry: {0}", inquiry.ToString()));
        }
    }
}
